sql/hive-thriftserver/v2.3/src/main/java/org/apache/hive/service/cli/operation/Operation.java

```diff
@@ -298,7 +298,10 @@ public void cancel() throws HiveSQLException {
     throw new UnsupportedOperationException("SQLOperation.cancel()");
   }
 
-  public abstract void close() throws HiveSQLException;
+  public void close() throws HiveSQLException {
+    setState(OperationState.CLOSED);
+    cleanupOperationLog();
+  }
```

Comment:
I needed to add this concrete base implementation here, otherwise the abstract override of close() in the trait didn't work in SparkExecuteStatementOperation, because it didn't have a concrete implementation in any of the subclasses.
On the other hand, if I wanted to implement it concretely in the trait, it was complaining about not being able to call protected methods from Java class in the trait.

This base implementation here is doing what it was doing in all the operations we have.
It might have been another tiny bug that SparkExecuteStatementOperation did not call cleanupOperationLog. The other operations did via their super call.